backend/src/main/java/com/google/minicrm/data/Settings.java

```diff
@@ -47,10 +87,27 @@ public Settings(Entity entity) {
         this.phone = (String) entity.getProperty("phone");
         this.phoneNotificationsFrequency =
                 NotificationsFrequency.fromDisplayed((String) entity.getProperty("phoneNf"));
-        this.language = Language.fromIsoCode((String) entity.getProperty("language"));
         this.currency = Currency.fromIsoCode((String) entity.getProperty("currency"));
     }
 
+    public static Settings fromReader(Reader reader, Key advertiserKey) throws NumberParseException, ValidatorException, MalformedJsonException {
+        Gson gson = new Gson();
+        Map<String, String> jsonMap = gson.fromJson(reader, Map.class);
+        if (!(jsonMap == null && jsonMap.containsKey("email") && jsonMap.containsKey("phone") &&
+                jsonMap.containsKey("emailNotificationsFrequency") &&
+                jsonMap.containsKey("phoneNotificationsFrequency") &&
+                jsonMap.containsKey("currency"))) {
+            throw new MalformedJsonException("One or more missing keys in JSON.");
+        }
+        String email = jsonMap.get("email");
```

Comment:
nit: Ok for now, maybe in the future we can configure GSON to custom deserialize the JSON for us.